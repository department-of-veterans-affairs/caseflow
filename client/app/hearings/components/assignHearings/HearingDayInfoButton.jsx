import PropTypes from 'prop-types';
import React from 'react';
import moment from 'moment';

import Tooltip from '../../../components/Tooltip';
import Button from '../../../components/Button';
import {
  buttonSelectedStyle,
  buttonUnselectedStyle,
  dateStyle,
  leftColumnStyle,
  rightColumnStyle,
  typeAndJudgeStyle,
  slotDisplayStyle,
  scheduledDisplayStyle
} from './styles';

import {
  vljFullnameOrEmptyString,
  formatSlotRatio,
  hearingDayHasJudge,
  separatorIfJudgeOrRoomPresent,
  hearingRoomOrEmptyString
} from '../../utils';

import { DOCKET_READABLE_REQUEST_TYPE } from '../../constants';

export const HearingDayInfoButton = ({ id, hearingDay, selected, onSelectedHearingDayChange }) => {
  // Format the pieces of information from hearingDay
  const formattedHearingType = DOCKET_READABLE_REQUEST_TYPE[hearingDay.requestType];
  const judgeOrRoom = hearingDayHasJudge(hearingDay) ?
    vljFullnameOrEmptyString(hearingDay) :
    hearingRoomOrEmptyString(hearingDay);
  const separator = separatorIfJudgeOrRoomPresent(hearingDay);
  const formattedSlotRatio = formatSlotRatio(hearingDay);
  const formattedDate = moment(hearingDay.scheduledFor).format('ddd MMM D');

  const classNames = selected ? ['selected-hearing-day-info-button'] : ['unselected-hearing-day-info-button'];

  return (
    <Button
      styling={selected ? buttonSelectedStyle : buttonUnselectedStyle}
      onClick={() => onSelectedHearingDayChange(hearingDay)}
      classNames={classNames}
      linkStyling
    >
      <div>
        <div {...leftColumnStyle} >
          <div {...dateStyle}>
            {formattedDate}
          </div>
          <div {...typeAndJudgeStyle}>
            <Tooltip id={`hearing-day-${id}`} text={judgeOrRoom} position="bottom" tabIndex={-1}>
              <span>{`${formattedHearingType} ${separator} ${judgeOrRoom}`}</span>
            </Tooltip>
          </div>
        </div>
        <div {...rightColumnStyle} >
          <div {...slotDisplayStyle}>
            {formattedSlotRatio}
          </div>
          <div {...scheduledDisplayStyle}>scheduled</div>
        </div>
      </div>
    </Button>
  );
};

HearingDayInfoButton.propTypes = {
  id: PropTypes.number,
  hearingDay: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  selected: PropTypes.bool,
  onSelectedHearingDayChange: PropTypes.func,
};

export default HearingDayInfoButton;
