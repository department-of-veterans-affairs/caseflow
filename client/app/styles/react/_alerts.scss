
.usa-alert {

  &.fixed {
    position: fixed;
    top: 0;
    margin-top: 0;
    z-index: 999;
    left: 50%;
    transform: translate(-50%, 0);
  }

  // Remove after USWDS V.2.0
  &.usa-alert-slim {
    background-size: 4rem;

    .usa-alert-body {
      padding-left: 3.5rem;
    }

    .usa-alert-text {
      font-size: 2rem;
    }
  }

  .usa-alert-text {
    line-height: 1.5;
  }
}
