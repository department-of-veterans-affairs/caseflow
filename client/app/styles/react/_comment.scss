.comment-control-button-container {
  display: flex;
  width: 100%;
  padding: 3px 0;
  justify-content: space-between;
}

.comment-save-button-container {
  display: inline-block;
  width: 100%;
  margin-top: 5px;
  margin-bottom: 17px;
}

.comment-control-button {
  padding: 0;

  &.horizontal {
    text-align: left;
  }
}

.comment-control-button-divider {
  padding: 0 10px;
}

.comment-container-selected {
  box-shadow: 3px 3px 5px $color-gray-light;
}

.comment-container {
  word-wrap: break-word;
  word-break: break-word;
  background-color: $color-white;
  padding: 15px;
  cursor: pointer;

  &:not(.comment-horizontal-container) {
    margin-bottom: 15px;
    border-radius: 3px;
    border: 1px solid $color-gray-lighter;
  }

  &.comment-horizontal-container {
    padding: 25px 25px 25px 40px;
  }
}

// Max width is enforced so the size of this comment box
// fits with existing textareas in style guide
.comment-size-container {
  max-width: 430px;

  .cf-form-textarea {
    margin-bottom: 20px;
  }
}

.comment-horizontal-container {
  display: inline-block;
  width: 100%;
}

.comment-horizontal-spacing {
  margin-right: 49px;
}

.commentIcon-container {
  position: absolute;
  z-index: 10;
  // These dimensions need to match the comment icon SVG dimensions.
  width: 40px;
  height: 40px;
}

