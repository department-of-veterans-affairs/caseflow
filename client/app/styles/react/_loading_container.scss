.loadingContainer-positioning {
  position: absolute;
  width: 100%;
  height: 100%;
}

.loadingContainer-table {
  width: 100%;
  height: 100%;
  display: table;
}

.loadingContainer-table-cell {
  text-align: center;
  display: table-cell;
  vertical-align: middle;
}

.loadingContainer-content {
  position: relative;
  z-index: 1;

  iframe {
    width: 100%;
    height: 80vh;

    &.form9Viewer {
      height: 500px;
    }
  }
}

.loadingContainer-container {
  position: relative;
  overflow: hidden;
}

.cf-loading-button-text {
  display: inline-block;
  margin-right: 10px;
  line-height: 0;
  position: relative;
}

.cf-loading-message {
  .loading-message-text {
    display: inline-block;
    font-style: italic;
    font-weight: bolder;
  }

}

.cf-loading-button-symbol {
  display: inline-block;
}
