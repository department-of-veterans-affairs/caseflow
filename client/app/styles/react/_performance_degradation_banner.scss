.usa-banner {
  background-color: $color-gold-lightest;
  text-align: center;
}

.usa-banner-inner {
  padding: .8em;
}

.banner-icon {
  display: inline-block;
  vertical-align: text-top;
  // height: 25px;
}

.banner-text {
  padding-left: 6px;
}
