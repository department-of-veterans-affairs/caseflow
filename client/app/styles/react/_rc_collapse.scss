@mixin common() {

  width: 0;
  height: 0;
  font-size: 0;
  line-height: 0;
}

@mixin right($w, $h, $color) {

  border-top: $w solid transparent;
  border-bottom: $w solid transparent;
  border-left: $h solid $color;
}

@mixin bottom($w, $h, $color) {

  border-left: $w solid transparent;
  border-right: $w solid transparent;
  border-top: $h solid $color;
}

// scss-lint:disable IdSelector
#arrow {
  @include common();
  @include right(3px, 4px, $color-gray);
  @include bottom(3px, 4px, $color-gray);
}

.rc-collapse {

  &-anim-active {
    transition: height .2s ease-out;
  }

  & > &-item {
    &:first-child {
      border-top: none;
    }
  }

  & > &-item-disabled > .rc-collapse-header {
    cursor: not-allowed;
    color: $color-gray-light;
    background-color: $color-gray-lighter;
  }

  &-content {
    overflow: hidden;

    &-inactive {
      display: none;
    }
  }

  &-item:last-child {
    > .rc-collapse-content {
      border-radius: 0 0 3px 3px;
    }
  }

  & > &-item-active {
    > .rc-collapse-header {

      .arrow {

        #arrow > {
          @include bottom(3px, 4px, $color-gray);
        }
      }
    }
  }
}

.usa-accordion-content {
  &:not([aria-hidden]) {
    position: relative;
    left: 0;
  }
}
