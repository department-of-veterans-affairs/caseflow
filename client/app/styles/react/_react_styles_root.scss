@import '_comment.scss';
@import '_search_bar.scss';
@import '_performance_degradation_banner.scss';
@import '_rc_collapse.scss';
@import '_loading_container.scss';
@import '_select-dropdown.scss';
@import '_toggle_button.scss';
