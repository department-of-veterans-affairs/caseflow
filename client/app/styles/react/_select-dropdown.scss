$cf-select-bg-color: #fff;
$cf-select-disabled-bg-color: $color-gray-lightest;
$cf-select-disabled-color: #323a45;
$cf-select-border-color: #5b616b;
$cf-select-text-color: #323a45;

.cf-select {
  .cf-select__control {
    background-color: $cf-select-bg-color;
    border: 1px solid $cf-select-border-color;
    border-collapse: separate;
    border-radius: 0;
    border-spacing: 0;
    color: $cf-select-text-color;
    cursor: default;

    &--is-disabled {
      background-color: $cf-select-disabled-bg-color;

      .cf-select__single-value {
        color: $cf-select-disabled-color;
      }

      .cf-select__indicators {
        opacity: 0.35;
      }

      input {
        margin: 0;
      }
    }

    .cf-select__value-container {
      padding-bottom: 0;
      padding-top: 0;

      .cf-select__placeholder {
        color: $cf-select-text-color;
      }

      // Ensure that the "selected" value is aligned correctly
      .cf-select__placeholder,
      .cf-select__single-value {
        position: unset;
        top: unset;
        transform: unset;
      }
    }

    .cf-select__indicators {
      .cf-select__indicator-separator {
        background-color: $cf-select-border-color;
        margin-bottom: 0;
        margin-top: 0;
      }

      .cf-select__indicator {
        svg {
          fill: $cf-select-text-color;
        }
      }
    }
  }

  .cf-select__menu {
    border-radius: 0;

    .cf-select__menu-list {
      .cf-select__option {
        background-color: $color-white;
        color: $color-gray;

        &.cf-select__option--is-focused {
          background-color: $color-gray;
          color: $color-white;
        }
      }
    }
  }

  .usa-input-error & {
    .cf-select--is-disabled {
      input { width: 1px; }
    }
  }
}
