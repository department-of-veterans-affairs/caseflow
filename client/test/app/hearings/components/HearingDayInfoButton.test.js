import React from 'react';

import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { axe } from 'jest-axe';

import HearingDayInfoButton from 'app/hearings/components/assignHearings/HearingDayInfoButton';

describe('HearingDayInfoButton', () => {
  const setSelectedValue = jest.fn();

  const props = {
    hearingDay:
      {
        scheduledFor: '04/12/2021',
        judgeFirstName: 'Jonas',
        judgeLastName: 'Jerengal',
        room: '14',
        requestType: 'V',
        readableRequestType: 'video',
        hearings: {
          hearing_1: {},
          hearing_2: {},
        },
        totalSlots: 8,
      },
    selected: false,
    onSelectedHearingDayChange: setSelectedValue,
  };

  it('renders correctly', () => {
    const { container } = render(<HearingDayInfoButton {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('passes a11y testing', async () => {
    const { container } = render(<HearingDayInfoButton {...props} />);

    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });

  it('has correct class when selected', () => {
    const selectedProps = { ...props, selected: true };
    const utils = render(<HearingDayInfoButton {...selectedProps} />);
    const button = utils.getByRole('button');

    expect(button).toHaveClass('selected-hearing-day-info-button');
  });

  it('has correct class when unselected', () => {
    const selectedProps = { ...props, selected: false };
    const utils = render(<HearingDayInfoButton {...selectedProps} />);
    const button = utils.getByRole('button');

    expect(button).toHaveClass('unselected-hearing-day-info-button');
  });

  it('calls setSelected on click', async () => {
    const utils = render(<HearingDayInfoButton {...props} />);
    const button = utils.getByRole('button');

    expect(setSelectedValue).toHaveBeenCalledTimes(0);
    await userEvent.click(button);
    expect(setSelectedValue).toHaveBeenCalledTimes(1);
  });

  it('renders the Docket readable request type when Docket is type Virtual', () => {
    props.requestType = 'R';

    const { container } = render(<HearingDayInfoButton {...props} />);

    expect(container).toMatchSnapshot();
  });
});
