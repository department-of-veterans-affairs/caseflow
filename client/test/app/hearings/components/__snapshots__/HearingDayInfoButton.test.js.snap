// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HearingDayInfoButton renders correctly 1`] = `
<div>
  <span>
    <button
      class="unselected-hearing-day-info-button cf-btn-link usa-button"
      data-css-1fa3cif=""
      type="button"
    >
      <div>
        <div
          data-css-16v9b8i=""
        >
          <div
            data-css-1o3wy7o=""
          >
            Mon Apr 12
          </div>
          <div
            data-css-5bhwlq=""
          >
            <span
              aria-describedby="hearing-day-undefined"
              aria-label=""
              currentitem="false"
              data-event="focus mouseenter"
              data-event-off="mouseleave keydown"
              data-for="hearing-day-undefined"
              data-tip="true"
              tabindex="-1"
            >
              Video · VLJ Jerengal, Jonas
            </span>
            <span
              data-css-10mgax4=""
            >
              <div
                class="__react_component_tooltip t00000000-0000-0000-0000-000000000000 place-bottom type-dark"
                data-id="tooltip"
                id="hearing-day-undefined"
                role="tooltip"
              >
                <style>
                  
  	.t00000000-0000-0000-0000-000000000000 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t00000000-0000-0000-0000-000000000000.place-top {
        margin-top: -10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-top::before {
        border-top: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-bottom {
        margin-top: 10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-left {
        margin-left: -10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-left::before {
        border-left: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-right {
        margin-left: 10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-right::before {
        border-right: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
                </style>
                VLJ Jerengal, Jonas
              </div>
            </span>
          </div>
        </div>
        <div
          data-css-1fnaqki=""
        >
          <div
            data-css-lrak98=""
          >
            0 of 8
          </div>
          <div
            data-css-heqg99=""
          >
            scheduled
          </div>
        </div>
      </div>
    </button>
  </span>
</div>
`;

exports[`HearingDayInfoButton renders the Docket readable request type when Docket is type Virtual 1`] = `
<div>
  <span>
    <button
      class="unselected-hearing-day-info-button cf-btn-link usa-button"
      data-css-1fa3cif=""
      type="button"
    >
      <div>
        <div
          data-css-16v9b8i=""
        >
          <div
            data-css-1o3wy7o=""
          >
            Mon Apr 12
          </div>
          <div
            data-css-5bhwlq=""
          >
            <span
              aria-describedby="hearing-day-undefined"
              aria-label=""
              currentitem="false"
              data-event="focus mouseenter"
              data-event-off="mouseleave keydown"
              data-for="hearing-day-undefined"
              data-tip="true"
              tabindex="-1"
            >
              Video · VLJ Jerengal, Jonas
            </span>
            <span
              data-css-10mgax4=""
            >
              <div
                class="__react_component_tooltip t00000000-0000-0000-0000-000000000000 place-bottom type-dark"
                data-id="tooltip"
                id="hearing-day-undefined"
                role="tooltip"
              >
                <style>
                  
  	.t00000000-0000-0000-0000-000000000000 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t00000000-0000-0000-0000-000000000000.place-top {
        margin-top: -10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-top::before {
        border-top: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-bottom {
        margin-top: 10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-left {
        margin-left: -10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-left::before {
        border-left: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-right {
        margin-left: 10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-right::before {
        border-right: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
                </style>
                VLJ Jerengal, Jonas
              </div>
            </span>
          </div>
        </div>
        <div
          data-css-1fnaqki=""
        >
          <div
            data-css-lrak98=""
          >
            0 of 8
          </div>
          <div
            data-css-heqg99=""
          >
            scheduled
          </div>
        </div>
      </div>
    </button>
  </span>
</div>
`;
